splitio/models/telemetry.py

```diff
@@ -18,72 +18,115 @@
-BLOCK_UNTIL_READY_TIMEOUT = 'blockUntilReadyTimeout'
-NOT_READY = 'notReady'
-TIME_UNTIL_READY = 'timeUntilReady'
-REFRESH_RATE = 'refreshRate'
-URL_OVERRIDE = 'urlOverride'
-HTTP_PROXY = 'httpProxy'
-
-HTTP_LATENCIES = 'httpLatencies'
-METHOD_LATENCIES = 'methodLatencies'
-METHOD_EXCEPTIONS = 'methodExceptions'
-LAST_SYNCHRONIZATIONS = 'lastSynchronizations'
-HTTP_ERRORS = 'httpErrors'
-STREAMING_EVENTS = 'streamingEvents'
-SPLIT = 'split'
-SEGMENT = 'segment'
-IMPRESSION = 'impression'
-IMPRESSION_COUNT = 'impressionCount'
-EVENT = 'event'
-TELEMETRY = 'telemetry'
-TOKEN = 'token'
-TREATMENT = 'treatment'
-TREATMENTS = 'treatments'
-TREATMENT_WITH_CONFIG = 'treatmentWithConfig'
-TREATMENTS_WITH_CONFIG = 'treatmentsWithConfig'
-TRACK = 'track'
-
-class StorageType(object):
-    """
-    Storage types constants
-
-    """
-    MEMEORY = 'memory'
+class CounterConstants(Enum):
+    """Impressions and events counters constants"""
+    IMPRESSIONS_QUEUED = 'impressionsQueued'
+    IMPRESSIONS_DEDUPED = 'impressionsDeduped'
+    IMPRESSIONS_DROPPED = 'impressionsDropped'
+    EVENTS_QUEUED = 'eventsQueued'
+    EVENTS_DROPPED = 'eventsDropped'
+
+class ConfigParams(Enum):
+    """Config parameters constants"""
+    SPLITS_REFRESH_RATE = 'featuresRefreshRate'
+    SEGMENTS_REFRESH_RATE = 'segmentsRefreshRate'
+    IMPRESSIONS_REFRESH_RATE = 'impressionsRefreshRate'
+    EVENTS_REFRESH_RATE = 'eventsPushRate'
+    TELEMETRY_REFRESH_RATE = 'metricsRefreshRate'
+    OPERATION_MODE = 'operationMode'
+    STORAGE_TYPE = 'storageType'
+    STREAMING_ENABLED = 'streamingEnabled'
+    IMPRESSIONS_QUEUE_SIZE = 'impressionsQueueSize'
+    EVENTS_QUEUE_SIZE = 'eventsQueueSize'
+    IMPRESSIONS_MODE = 'impressionsMode'
+    IMPRESSIONS_LISTENER = 'impressionListener'
+
+class ExtraConfig(Enum):
+    """Extra config constants"""
+    ACTIVE_FACTORY_COUNT = 'activeFactoryCount'
+    REDUNDANT_FACTORY_COUNT = 'redundantFactoryCount'
+    BLOCK_UNTIL_READY_TIMEOUT = 'blockUntilReadyTimeout'
+    NOT_READY = 'notReady'
+    TIME_UNTIL_READY = 'timeUntilReady'
+    REFRESH_RATE = 'refreshRate'
+    HTTP_PROXY = 'httpProxy'
+    HTTPS_PROXY_ENV = 'HTTPS_PROXY'
+
+class ApiURLs(Enum):
+    """Api URL constants"""
+    SDK_URL = 'sdk_url'
+    EVENTS_URL = 'events_url'
+    AUTH_URL = 'auth_url'
+    STREAMING_URL = 'streaming_url'
+    TELEMETRY_URL = 'telemetry_url'
+    URL_OVERRIDE = 'urlOverride'
+
+class HTTPExceptionsAndLatencies(Enum):
+    """Sync exceptions and latencies constants"""
+    HTTP_ERRORS = 'httpErrors'
+    HTTP_LATENCIES = 'httpLatencies'
+    SPLIT = 'split'
+    SEGMENT = 'segment'
+    IMPRESSION = 'impression'
+    IMPRESSION_COUNT = 'impressionCount'
+    EVENT = 'event'
+    TELEMETRY = 'telemetry'
+    TOKEN = 'token'
+
+class MethodExceptionsAndLatencies(Enum):
+    """Method exceptions and latencies constants"""
+    METHOD_LATENCIES = 'methodLatencies'
+    METHOD_EXCEPTIONS = 'methodExceptions'
```

Comment:
i'd say these 2 should be separate constants, or at least not be mixed with the method names

Reply:
They both use the same exact method names, that is why I had them in the same class